pmd-java/etc/grammar/Java.jjt

```diff
@@ -1897,11 +1950,17 @@ Token t;}
 void ExplicitConstructorInvocation() :
 {}
 {
-  LOOKAHEAD("this" Arguments() ";") "this" {jjtThis.setIsThis();} Arguments() ";"
+  LOOKAHEAD("this" ArgumentList() ";") "this" ArgumentList() ";"
```

Comment:
In fact we can optimise the hell out of this, and only look ahead one token:
```java
  ( LOOKAHEAD("<") TypeArguments() ("this" | "super" {jjtThis.setIsSuper();})
  | LOOKAHEAD("this") "this"
  | LOOKAHEAD("super") "super" {jjtThis.setIsSuper();}
  | PrimaryExpression() "." [ TypeArguments() ] "super" {jjtThis.setIsSuper();}
  )
  ArgumentList() ";"
```
This is because 
1. `this` or `super` can't appear in the primary expression in case of qualified super call, or else compile error, so if we see them, then the rest is necessarily the arguments
2. No primary expression may start with `<`, meaning we can use it to know we're looking forward to type arguments